WireDomain/Tests/WireDomainTests/Repositories/ConversationLabelsRepositoryTests.swift

```diff
@@ -0,0 +1,332 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+@testable import WireAPI
+import WireAPISupport
+@testable import WireDataModel
+import WireDataModelSupport
+@testable import WireDomain
+import XCTest
+
+final class ConversationLabelsRepositoryTests: XCTestCase {
+
+    var sut: ConversationLabelsRepository!
+    var userPropertiesAPI: MockUserPropertiesAPI!
+
+    var stack: CoreDataStack!
+    let coreDataStackHelper = CoreDataStackHelper()
+    let modelHelper = ModelHelper()
+
+    private var conversation1: ZMConversation!
+    private var conversation2: ZMConversation!
+    private var conversation3: ZMConversation!
+
+    var context: NSManagedObjectContext {
+        stack.syncContext
+    }
+
+    override func setUp() async throws {
+        try await super.setUp()
+        /// Batch requests don't work with in-memory store
+        /// so we need to use a persistent store.
+        stack = try await coreDataStackHelper.createStack(inMemoryStore: false)
```

Comment:
yes it is already done by the CoreDataStackHelper using a random UUID so at every test we're creating a new store at a specific `temporary` location (not removing it though) which should be enough to have the tests passing but they don't anymore

Reply:
So, I added some clean up of the temp directory so we make sure to remove the directory after each test however for some reason we still need to clean up the entity so we have a clean database and have the tests pass. Not sure why tbh, I tried removing the file, closing the stores (that do some reset of the context), it would not work. It behaves in a weird way, I figured cleaning up the entity is the most straightforward approach.

Reply:
you can check in last commit

Reply:
Maybe we should recreate the `CoreDataStackHelper()` in the setup too. I looked quickly at the implementation and the location is unique per `CoreDataStackHelper` instance. That could be causing the issues you mentioned above.

Reply:
yeah did that too, still not working

Reply:
Strange. I still think it is worth creating `CoreDataStackHelper` in setup though.

Reply:
fixed